src/main/java/roboy/dialog/states/expoStates/RoboyInfoState.java

```diff
@@ -0,0 +1,207 @@
+package roboy.dialog.states.expoStates;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import roboy.context.Context;
+import roboy.context.contextObjects.IntentValue;
+import roboy.dialog.states.definitions.State;
+import roboy.dialog.states.definitions.StateParameters;
+import roboy.linguistics.Linguistics.UtteranceSentiment;
+import roboy.linguistics.sentenceanalysis.Interpretation;
+import roboy.memory.nodes.Interlocutor;
+import roboy.memory.nodes.Roboy;
+import roboy.talk.PhraseCollection;
+import roboy.util.QAJsonParser;
+import roboy.util.RandomList;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.ZoneId;
+import java.time.LocalDate;
+import java.time.Period;
+import java.util.*;
+
+import static roboy.memory.Neo4jProperty.*;
+
+public class RoboyInfoState extends State {
+    public final static String INTENTS_HISTORY_ID = "RIS";
+
+    private final String SELECTED_SKILLS = "skills";
+    private final String SELECTED_ABILITIES = "abilities";
+    private final String SELECTED_ROBOY_QA = "roboy";
+    private final String LEARN_ABOUT_PERSON = "newPerson";
+    private final String INFO_FILE_PARAMETER_ID = "infoFile";
+    private final Logger LOGGER = LogManager.getLogger();
```

Comment:
many transitions defined here
getRequiredTransitions() function should be implemented to detect missing definitions in the personality file and simplify debugging

also, this state is not modular because it assumes that the transitions point to specific states

Reply:
Removed